### PR TITLE
Merge 7 -> 8

### PR DESCRIPTION
**Pull request by iche033**
# ➡️ Forward port

Port gz-sim7 to gz-sim8

Branch comparison: https://github.com/gazebosim/gz-sim/compare/gz-sim8...gz-sim7

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by iche033** [comment]
This needs an sdformat14 release.

**Comment by iche033** [comment]
> This needs an sdformat14 release.

sdformat 14.5.0 is out. CI works now

**Review by nkoenig — APPROVED** [review]

File: src/SimulationRunner.cc [hunk 1]
```diff
@@ -90,40 +94,24 @@ struct MaybeGilScopedRelease
...
-  this->worldName = transport::TopicUtils::AsValidTopic(_world->Name());
-
-  auto validWorldName = transport::TopicUtils::AsValidTopic(worldName);
-  if (this->worldName.empty())
-  {
-    gzerr << "Can't start simulation runner with this world name ["
-          << worldName << "]." << std::endl;
-    return;
-  }
+  this->worldName = _world.Name();
```
[the remark this answers is not in the document]
Comment (iche033) [reply src/SimulationRunner.cc:103]: fixed.

**Comment by azeey** [comment]
Commented on a change of behavior in https://github.com/gazebosim/gz-sim/pull/2452.

**Comment by azeey** [comment]
@iche033 I've just merged #2527. Can you update this PR to include it?

**Comment by iche033** [comment]
merged from gz-sim7 and removed deprecations in 8bb720883c2f3954bb140fc25b41965830072776

**Review by azeey — APPROVED** [review]

**Pull request merged**